### PR TITLE
adding a Rebar object (cross-sectional)

### LINKED ISSUE #268: [✨ Feature request]: EPIC: Rectangular Reinforced Cross Section
### Description

Implement a Rectangular Reinforced Cross Section. In the future to be used with a separate Analysis class.

### Acceptance Criteria
- [x] Concrete Material;
- [x]  Rebar Material;
- [ ]  Rebar object (material, location etc.);
- [ ]  (Reinforced) CrossSection Abstract Baseclass or Protocol (area, perimeter, list of nodes?, section moduli, rebars, cover_definitions);
- [ ]  ReinforcedRectangularCrossSection definition; 

### Dependencies

_No response_

### Confirmation

- [ ]  I have checked that this feature does not already exist in the issue tracker.
- [ ]  I have read and understood the contribution guidelines.

### PR DESCRIPTION
**Pull request by egarciamendez**
## Description

Adding a Rebar object (cross-sectional)

Fixes #268 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: blueprints/structural_sections/concrete/rebar.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    def __init__(
+        self,
+        diameter: MM,
+        x: MM,
+        y: MM,
+        material: ReinforcementSteelMaterial,
+        relative_start_position: RATIO = 0.0,
+        relative_end_position: RATIO = 1.0,
+    ) -> None:
+        """Initialize the Rebar object."""
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/structural_sections/concrete/rebar.py]: done

File: blueprints/structural_sections/concrete/rebar.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        """Diameter of the rebar [mm]."""
+        return self._diameter
+
+    @diameter.setter
+    def diameter(self, value: MM) -> None:
+        """Set the diameter of the rebar."""
+        if value <= 0.0:
+            msg = "The diameter of the rebar must be greater than zero"
+            raise ValueError(msg)
+        self._diameter = value
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/structural_sections/concrete/rebar.py]: it is now

File: blueprints/structural_sections/concrete/reinforced_concrete_sections/cross_section_shapes.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import math
+from dataclasses import dataclass, field
+from typing import Protocol
+
+from shapely import Point, Polygon
+
+from blueprints.type_alias import MM, MM2
+
+
+@dataclass
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/structural_sections/concrete/reinforced_concrete_sections/cross_section_shapes.py]: You're right, thanks!

File: blueprints/structural_sections/concrete/reinforced_concrete_sections/cross_section_shapes.py [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+        y : MM
+            The y-coordinate of the point [mm].
+
+        Returns
+        -------
+        bool
+            True if the point is inside the circle, False otherwise.
+        """
+        # Check if the point is within the circular polygon
+        return self.circle.contains(Point(x, y))
```
[the remark this answers is not in the document]
Comment (egarciamendez) [reply blueprints/structural_sections/concrete/reinforced_concrete_sections/cross_section_shapes.py]: better to skip contains_point. Shapely has already all that kind of operations.

File: blueprints/structural_sections/concrete/rebar.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    def __post_init__(self) -> None:
+        """Post-initialization to validate the diameter."""
+        super().__post_init__()
+        if not 0.0 <= self.relative_end_position <= 1.0:
+            msg = f"Relative start position of the rebar must be between 0.0 and 1.0, but got {self.relative_end_position}"
+            raise ValueError(msg)
+
+        if not 0.0 <= self.relative_start_position <= 1.0:
+            msg = f"Relative start position of the rebar must be between 0.0 and 1.0, but got {self.relative_start_position}"
+            raise ValueError(msg)
```
Comment (SZeltaat) [inline blueprints/structural_sections/concrete/rebar.py:41]:
```suggestion
    def __post_init__(self) -> None:
        """Post-initialization to validate the diameter."""
        super().__post_init__()
        if not 0.0 <= self.relative_end_position <= 1.0:
            msg = f"Relative end position of the rebar must be between 0.0 and 1.0, but got {self.relative_end_position}"
            raise ValueError(msg)

        if not 0.0 <= self.relative_start_position <= 1.0:
            msg = f"Relative start position of the rebar must be between 0.0 and 1.0, but got {self.relative_start_position}"
            raise ValueError(msg)
```

**Review by SZeltaat — APPROVED** [review]
Great!
There's only one typo. The rest is ready to 🚢🚀

**Pull request merged**